### PR TITLE
Cleanup and Testing

### PR DESCRIPTION
**Pull request by ZachPeterson**
Cleaned up a few things with the database and models. Added some default testing values to be inserted into the database if in testing mode.

### CODE REVIEW

**Comment by goutamvenkat** [comment]
can we change the professor url request to not return a string version of bool? string comparisons are too slow

File: ClassMoodApp/Controllers/ClassList.py [hunk 1]
```diff
@@ -1,39 +1,34 @@
...
+    authUser = API.get_authentication()
     if not authUser:
         return render_template('authentication/login.html', error='You are not logged in')
-    professorEmail = authUser.email
-    successful = api.create_prof_lecture(className, professorEmail)
+    successful = API.create_prof_class(className, authUser.id)
     if successful:
         return 'True'
     else:
         return 'False'
```
Comment (goutamvenkat) [inline ClassMoodApp/Controllers/ClassList.py]: This is unnecessary, instead use `return json.dumps(successful)`

File: ClassMoodApp/static/js/controllers.js [hunk 2]
```diff
@@ -73,9 +74,9 @@ app.controller('dialogController', function($scope, $log, $http) {
         var className = $scope.classname;
         var studentID = 1;
         if (className !== '') {
-            $http.post('/set_student_lectures/' + className + '/' + studentID)
+            $http.post('/set_student_classes/' + className + '/' + studentID)
             .then(function successfulCallback(response) {
-                if (response.data === true) {
+                if (response.data == 'true') {
```
Comment (goutamvenkat) [inline ClassMoodApp/static/js/controllers.js]: No this is incorrect, don't use string comparison. Update the API and this function

File: ClassMoodApp/static/js/controllers.js [hunk 3]
```diff
@@ -49,10 +50,10 @@ app.controller('dialogProfessorController', function($scope, $log, $http) {
 });
 
 
-app.controller("studentLectures", function($scope, $http) {
+app.controller("studentClassListController", function($scope, $http) {
     // Get studentID from session token
     var studentID = 1;
```
Comment (goutamvenkat) [inline ClassMoodApp/static/js/controllers.js]: studentID is hardcoded. Needs to be retrieved from cookie

File: ClassMoodApp/Models/DBModels.py [hunk 4]
```diff
@@ -9,80 +9,77 @@
...
         self.created_at = datetime.now()
         self.expires_at = self.created_at + timedelta(hours=2)
     def __repr__(self):
         return "token:{}, id:{}, userid:{}, create:{}, expire:{}".format(str(self.token), str(self.id), str(self.user_id), str(self.created_at), str(self.expires_at))
 
-class Roster(db.Model):
-    __tablename__ = 'Roster'
-    student_id = db.Column(db.BigInteger(), db.ForeignKey('User.id'), primary_key=True)
-    lecture_id = db.Column(db.BigInteger(), db.ForeignKey('Lecture.id'), primary_key=True)
+class ClassMember(db.Model):
```
Comment (goutamvenkat) [inline ClassMoodApp/Models/DBModels.py]: Why was this table name changed??? Roster made more sense

**Review by goutamvenkat — CHANGES_REQUESTED** [review]
The changes I mentioned should be addressed before merging. Personally, I don't see why the table names were changed, but whatever. Regardless, there some places were there's no underscore casing, those should be fixed for consistent stylistic purposes. The main thing to change though is the api function that returns successful.

Besides that it seems fine.

**Comment by goutamvenkat** [comment]
In login page, material design lite doesn't work and also a bunch of js files don't load. 
Screenshot of console

**Comment by goutamvenkat** [comment]
Getting this after logging in: 
After login image

**Comment by goutamvenkat** [comment]
Once I'm logged in and try hitting localhost again, it sends me to this page
image

which makes absolutely no sense. The logout should be placed in each of the views (student and professor). If the user is authenticated, why are we redirecting to authtest.html?

**Review by krabagobanjo — CHANGES_REQUESTED** [review]
The authentication redirect is a known issue - I didn't want to screw with the UI for the professor and student pages but we needed some way to log out.  The framework is there for implementation and should be done for next sprint.

I attached one comment, but everything seems fine otherwise.

File: ClassMoodApp/static/js/controllers.js (hunk 2 above)
Comment (ZachPeterson) [inline ClassMoodApp/static/js/controllers.js]: Unfortunately this is correct. Angular documentation shows that the response data is held as a string (https://docs.angularjs.org/api/ng/service/$http) and logging the typeof response.data confirms this. We can either do a string comparison here or use the transform method to convert the string response from the server to a boolean.

File: ClassMoodApp/Models/DBModels.py (hunk 4 above)
Comment (ZachPeterson) [inline ClassMoodApp/Models/DBModels.py]: Table name changed to reflect the name agreed upon by the team during the database architecture meeting.

**Review by krabagobanjo — APPROVED** [review]

**Review by neilgoel — APPROVED** [review]

**Pull request merged**